centreon/src/Core/Security/Authentication/Infrastructure/Api/Login/SAML/CallbackController.php

```diff
@@ -0,0 +1,91 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Security\Authentication\Infrastructure\Api\Login\SAML;
+
+use Centreon\Application\Controller\AbstractController;
+use Core\Application\Common\UseCase\ErrorAuthenticationConditionsResponse;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\UnauthorizedResponse;
+use Core\Infrastructure\Common\Api\HttpUrlTrait;
+use Core\Security\Authentication\Application\UseCase\Login\ErrorAclConditionsResponse;
+use Core\Security\Authentication\Application\UseCase\Login\Login;
+use Core\Security\Authentication\Application\UseCase\Login\LoginRequest;
+use Core\Security\Authentication\Application\UseCase\Login\LoginResponse;
+use Core\Security\Authentication\Application\UseCase\Login\PasswordExpiredResponse;
+use Core\Security\Authentication\Domain\Exception\AuthenticationException;
+use FOS\RestBundle\View\View;
+use OneLogin\Saml2\Error;
+use OneLogin\Saml2\ValidationError;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+class CallbackController extends AbstractController
+{
+    use HttpUrlTrait;
+
+    /**
+     * @param Request $request
+     * @param Login $useCase
+     * @param CallbackPresenter $presenter
+     * @param SessionInterface $session
+     * @return object
+     * @throws AuthenticationException
+     */
+    public function __invoke(
+        Request $request,
+        Login $useCase,
+        CallbackPresenter $presenter,
+        SessionInterface $session
+    ): object
+    {
+        $samlLoginRequest = LoginRequest::createForSAML($request->getClientIp());
+
+        $useCase($samlLoginRequest, $presenter);
+
+        switch (true) {
+            case is_a($presenter->getResponseStatus(), PasswordExpiredResponse::class)
+                || is_a($presenter->getResponseStatus(), UnauthorizedResponse::class)
+                || is_a($presenter->getResponseStatus(), ErrorResponse::class):
+                return View::createRedirect(
+                    $this->getBaseUrl() . '/login?authenticationError=' . $presenter->getResponseStatus()->getMessage()
+                );
+            case is_a($presenter->getResponseStatus(), ErrorAclConditionsResponse::class):
+            case is_a($presenter->getResponseStatus(), ErrorAuthenticationConditionsResponse::class):
```

Comment:
`instanceof` is far better than `is_a` when using object instances (`->getResponseStatus()`).
- more expressive
- faster (at least x2)

Inconsistency : "or" used `||` but multiple `case` in another location...
```suggestion
            case $presenter->getResponseStatus() instanceof PasswordExpiredResponse:
            case $presenter->getResponseStatus() instanceof UnauthorizedResponse:
            case $presenter->getResponseStatus() instanceof ErrorResponse:
                return View::createRedirect(
                    $this->getBaseUrl() . '/login?authenticationError=' . $presenter->getResponseStatus()->getMessage()
                );
            case $presenter->getResponseStatus() instanceof ErrorAclConditionsResponse:
            case $presenter->getResponseStatus() instanceof ErrorAuthenticationConditionsResponse:
```